neo/rawio/biocamrawio.py

```diff
@@ -122,10 +122,27 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop, strea
             i_start = 0
         if i_stop is None:
             i_stop = self._num_frames
-        if channel_indexes is None:
-            channel_indexes = slice(None)
+
+
         data = self._read_function(self._filehandle, i_start, i_stop, self._num_channels)
-        return data[:, channel_indexes]
+
+        # older style data return everything
+        if len(data.shape) > 1:
```

Comment:
Confused about how do you compare a tuple to > 1.

Reply:
It's the len of the tuple. Not the tuple itself.

Reply:
In a flat array the len of the shape will be 1 because it will be 

data.shape = (n_samples)

whereas the full array will be 2d

data.shape = (n_samples, n_channels)

Reply:
Ah, thanks, I did not see this, my bad. Maybe using ndim https://numpy.org/doc/stable/reference/generated/numpy.ndarray.ndim.html will make this clearear?

Reply:
I always forget ndim exists. Thanks that is what I wanted, but for the life of me couldn't remember that. So len(shape) does the same while being more unclear.

Reply:
Yes, same for me. There is size, shape and that other thing that... ehm.